dspace-api/src/main/java/org/dspace/identifier/DOIIdentifierProvider.java

```diff
@@ -832,7 +1089,39 @@ protected void removeDOIFromObject(Context context, DSpaceObject dso, String doi
 
         itemService.clearMetadata(context, item, MD_SCHEMA, DOI_ELEMENT, DOI_QUALIFIER, null);
         itemService.addMetadata(context, item, MD_SCHEMA, DOI_ELEMENT, DOI_QUALIFIER, null,
-                                remainder);
+                remainder);
         itemService.update(context, item);
     }
-}
+
+    /**
+     * Checks to see if an item can have a DOI minted, using the configured logical filter
+     * @param context
+     * @param dso The item to be evaluated
+     * @return
+     * @throws DOIIdentifierNotApplicableException
+     */
+    @Override
+    public Boolean canMint(Context context, DSpaceObject dso) throws DOIIdentifierNotApplicableException {
+        // Default is 'true' in the case of a null/missing filter. All we really care about is whether
+        // an exception was thrown or not.
+        log.debug("canMint is being called");
```

Comment:
Hi @kshepherd , sorry to jump into this discussion.
I understand the expected behaviour of this method as correctly explained in comments, and agree on what we really care about. 
But with boolean as returned value, what happens if a client of this method, unaware of its behavior, calls it and makes his/her logic or whatever by handling a false return value that actually is not returned? 
Thank you

Reply:
Hi @corrad82-4s , I see what you're getting at... what would be your suggestion? Should we make this methoid void instead of Boolean and perhaps make it clearer by renaming to `checkMintable` ?

Reply:
hi @kshepherd , 
yes, if there is no need to deal with returned value, and the purpose of this method is raise an exception if a dso is not mintable I would make it void and rename it in checkMintable

Reply:
thanks @corrad82-4s - sorry for the late reply to this.
i have refactored the method signature and usage here - it is now public void checkMintable() and does not return a value
if it looks good to you, please resolve this conversation